src/plugins/query_enhancements/public/datasets/s3_type.ts

```diff
@@ -247,3 +260,63 @@ const fetchTables = async (http: HttpSetup, path: DataStructure[]): Promise<Data
 
   return fetch(http, path, 'TABLE');
 };
+
+// Function to process the input and map types using the new SQL to OSD mapping
+export function mapResponseToFields(sqlOutput: SQLQueryResponse): DatasetField[] {
+  const datasetFields: DatasetField[] = [];
+
+  for (const row of sqlOutput.datarows) {
+    const [colName, dataType] = row;
+
+    // Stop processing once we hit the partition info row
+    if (colName === S3_PARTITION_INFO_COLUMN) {
+      break;
+    }
+
+    // Only include rows with valid types
+    if (dataType && dataType !== '') {
+      const sqlType = dataType as OPENSEARCH_SQL_FIELD_TYPES;
```

Comment:
The SQL Types should be defined here

Reply:
Updated here: [6d6a9c6](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8609/commits/6d6a9c6b6100e4ca696875e237fe793c07fd84a8)